src/Tools/dotnet-monitor/Exceptions/ExceptionsStore.cs

```diff
@@ -104,24 +105,85 @@ private async Task ProcessEntriesAsync(CancellationToken token)
                         moduleName = NameFormatter.GetModuleName(entry.Cache.NameCache, exceptionClassData.ModuleId);
                     }
 
+                    CallStackResultModel callStackResult = GenerateCallStackResult(entry.StackFrameIds, entry.Cache, entry.ThreadId);
+
                     lock (_instances)
                     {
-                        _instances.Add(new ExceptionInstance(exceptionTypeName, moduleName, entry.Message, entry.Timestamp));
+                        _instances.Add(new ExceptionInstance(exceptionTypeName, moduleName, entry.Message, entry.Timestamp, callStackResult));
                     }
                 }
 
                 shouldReadEntry = await _channel.Reader.WaitToReadAsync(token);
             }
         }
 
+        internal static CallStackResultModel GenerateCallStackResult(ulong[] stackFrameIds, IExceptionsNameCache cache, int threadId)
+        {
+            CallStackResult callStackResult = new CallStackResult();
+            CallStack callStack = new();
+            callStack.ThreadId = (uint)threadId;
+
+            foreach (var stackFrameId in stackFrameIds)
```

Comment:
Rather than iterating the name cache and inserting the data into another name cache, could you add another constructor to [CallStackData](https://github.com/dotnet/dotnet-monitor/blob/1e1bd44c6a049380b5c60cdcdca5a825858c1176/src/Microsoft.Diagnostics.Monitoring.WebApi/Stacks/CallStackData.cs#L12) that takes an existing NameCache? We're going to throw away this CallStackResult immediately after converting it to the model, so there shouldn't be any lifetime issues.